src/p5.dance.interpreted.js

```diff
@@ -56,25 +65,57 @@ function runUserSetup() {
   });
 }
 
+/**
+ * @param {Object} events - An object where each key is an event type. Each value
+ *  is another object, where the keys represeent the param of the event to be run
+ */
 function runUserEvents(events) {
-  var currentEvents = {};
+  // We have three separate event queues.
+  // First we run every N seconds/measures events
+  // Then we run after N seconds/measures events
+  // Finally we run all other events
+
+  var queues = {};
+  Object.keys(QueueType).forEach(function(key) {
+    queues[key] = [];
+  });
+
+  // Iterate through all of the inputEvents we've cached in the interpreter, looking
+  // to see if they meet the criteria of the passed in events object. If they do,
+  // we add them to the appropriate event queue
   for (var i = 0; i < inputEvents.length; i++) {
     var eventType = inputEvents[i].type;
-    var event = inputEvents[i].event;
     var param = inputEvents[i].param;
-    var priority = inputEvents[i].priority;
+    var queueType = inputEvents[i].queueType;
     if (events[eventType] && events[eventType][param]) {
-      //If there are multiple cues of the same type, only run the event with the highest priority
-      if (!currentEvents[eventType] || currentEvents[eventType].priority < priority) {
-        currentEvents[eventType] = {event: event, priority: priority};
+      if (!queues[queueType]) {
+        throw new Error('Unknown queueType: ', queueType);
       }
+      queues[queueType].push({
+        priority: inputEvents[i].priority,
+        func: inputEvents[i].func
+      });
     }
   }
-  for (var input in currentEvents){
-    if(currentEvents.hasOwnProperty(input)){
-      currentEvents[input].event();
-    }
+
+  function prioritySort(a, b) {
+    // TODO: If compareFunction(a, b) returns 0, leave a and b unchanged with respect
```

Comment:
Is *this* the compareFunction?  Slightly confused by what this comment refers to, and what is left TODO.

Reply:
Correct, `prioritySort` is passed to the `.sort` below:

https://github.com/code-dot-org/dance-party/blob/658c9a6334e1c47930efb050b5c95f65421f4612/src/p5.dance.interpreted.js#L116-L117

Chatted with Ryan, he's not worried about the "non-stable sort" case, because we only hit that if a student tries using two different hat blocks to run an event at the exact same time.  Any events with differing priority (or "after" vs. "every") will always be sorted correctly.